lib/df-detection.c

```diff
@@ -0,0 +1,286 @@
+#include "linux/df-detection.h"
+#include <asm/syscall.h>
+#include <linux/debugfs.h>
+#include <linux/fs.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+#include <linux/stacktrace.h>
+#include <linux/sys.h>
+#include <linux/types.h>
+
+void add_address(const void *addr, size_t len, unsigned long caller)
+{
+	if (current->addresses == NULL || current->pairs == NULL ||
+	    addr > TASK_SIZE)
+		return;
+	char buf[64], buf_caller[64];
+	int size;
+	size = scnprintf(buf, sizeof(buf), "%ps", caller);
+	/*TODO: find a better way to make kcov_ioctl not blocking syzkaller
+	 or i can make this filter in syzkaller*/
+	int size_caller =
+	    scnprintf(buf_caller, sizeof(buf_caller), "%ps", _RET_IP_);
+	/*ignore cases based on input(pointer and length)*/
+	if (len > MAX_LEN || addr == 0 ||
+	    strnstr(buf, "copy_from_user_nmi", size) ||
+	    strnstr(buf, "kcov_ioctl", size) ||
+	    strnstr(buf_caller, "kcov_ioctl", size_caller))
+		return;
+	if (current->num_read >= current->sz &&
+	    !WARN_ON(current->sz > DF_MAX_RECORDS)) {
```

Comment:
These warnings are causing syzkaller crashes and are not really actionable. Let's remove them and just skip the remaining addresses if we hit the limit.